app/src/main/java/edu/usf/sas/pal/muser/firebase/models/SongData.kt

```diff
@@ -0,0 +1,50 @@
+package edu.usf.sas.pal.muser.firebase.models
+
+
+/**
+ * A model class that holds the information about the song that is currently being played by the
+ * user.
+ */
+data class SongData
+
+/**
+ * [id] - The id of the track.
+ * [name] - Name of the track.
+ * [artistID] - The id of the artist.
+ * [artistName] - Name of the Artist who composed the track.
+ * [albumID] - The id of the album.
+ * [albumName] - Name of the album that the current track belongs to.
+ * [playlistID] - The id of the playlist that track belongs to.
+ * [playlistPlayOrder] - The queue number of the song in a playlist.
+ * [discNumber] - The disc number that the track belongs to.
+ * [bitrateLabel] - The bitrate of the song.
+ * [fileSizeLabel] - The file size of the song.
+ * [isPodCast] - Boolean value to determine if the current audio file is a pod cast or not.
+ * [duration] - The length of the track.
+ * [dateAdded] - The date on which the track was added.
+ * [year] - The year during which the track was released.
+ * [playCount] - The number of times the user played this track.
+ * [sampleRateLabel] - The sample rate of the song.
+ * [formatLabel] - The format of the song. (mp3 or Flac etc.)
+ */
+(val id: Long,
+ val name: String,
+ val artistID: Long,
+ val artistName: String,
+ val albumID: Long,
+ val albumName: String,
+ val playlistID: Long,
+ val playlistPlayOrder: Long?,
```

Comment:
We'll need to provide default values for anything that is nullable (has `?`) if we're not going to populate it in the constructor. We can use `""` for Strings (conversely, we could just make all the Strings non-nullable and always pass the value parsed from the tag if that value would always be `""`). Would `-1` for for numbers?

Going back to your earlier test:

>I tried logging what the ContentProvider would return for empty tags and here's what I got:-
2020-08-07 14:06:31.698 27838-27838/edu.usf.sas.pal.muser D/PlayerFragment: trackInfoChanged: <Unknown>

What exactly is the data type returned for empty tags in code (not the LogCat output)? I'm trying to figure out what the constructor will look like when we're creating these Firebase objects. It's actually probably best to go ahead and test that now - as an exercise, could you go ahead and add the code to instantiate one of these `SongData` objects with an event in the existing Shuttle code so we get an idea of what we're looking at?

Reply:
I think the tag `<unknown>` is returned to the `song.album` and the data type is `String`. 

I'll try capturing the `Event` object when the user pauses the track. Should I push the object to FireStore or should I log it locally in JSON format? I think the latter sounds better. Let me know your thoughts.

Reply:
You can go ahead and push to Firebase, we need to do that anyway. I think even just getting it to compile and run will answer some of these questions.